### PR TITLE
Update docs to cover QA tools and simplified conda environments

### LINKED ISSUE #390: Update docs to cover QA tools and simplified conda environments
With PR #381 merged, the docs should be updated to reflect the latest QA tool integrations and how to use them.

Additional sections to cover
- Update Development Environment process (#394)
- Version Control Workflow (now with pre-commit)
- Documentation

### PR DESCRIPTION
**Pull request by tomvothecoder**
## Description

This PR updates the docs to cover the QA tools and `pre-commit` (#381) and the simplified conda environments (#394).

Summary of Changes
- Update "Installation" -- "Installation in a conda environment" with simplified conda environments and `pre-commit`
- Update "Installation" page references for "Anvil/blues" to "LCRC" and "Cori" to "NERSC"
- Add "Developer Guide" -- "Project Standards" page to cover version control, QA tools, style guide, and documenting
- Remove "Developer Guide" --  "Creating a New Environment from Scratch" page -- the conda environments should not break with #385 and #394
  -  https://e3sm-project.github.io/e3sm_diags/docs/html/dev_guide/create-new-env.html#
- Add command line arg `--no-verify` to `git commit -m "..."` when contributing to docs
  - Since `pre-commit` installs hooks into the `.git` directory, it will complain about a Python error if `pre-commit` is not installed in the `sphinx` docs environment 
  -  Once docs are moved to `master` (#352), this no longer be needed

Fixes #
- Closes #390  

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

### CODE REVIEW

File: source/contributing.rst
Comment (tomvothecoder) [inline source/contributing.rst:98]: This is the main change for this file. The other changes were from PyCharm's formatting.

**Comment by tomvothecoder** [comment]
@chengzhuzhang @forsyth2 This PR is ready for review. I will push the docs build after approval. After this, I can start work on #352.

File: source/dev_guide/project-standards.rst
Comment (forsyth2) [inline source/dev_guide/project-standards.rst:32]: I would put these in the same tense (i.e., "Merging" and "Committing")

File: source/dev_guide/project-standards.rst
Comment (forsyth2) [inline source/dev_guide/project-standards.rst:87]: I would preface this list with something like "why squash commits?"

Perhaps it would also be good to add the caveat that if the PR has isolated commits then they can stay separated (although I suppose that point is made moot by the idea that a PR should only be handling one specific thing).

Another benefit is that it makes `git bisect` easier. (e.g., if you break the code in commit 1 and fix it in commit 2, then `git bisect` might get caught on commit 1 rather than the actual faulty commit).

File: source/dev_guide/project-standards.rst
Comment (forsyth2) [inline source/dev_guide/project-standards.rst:81]: You might want to add an example of squashing commits. 
e.g.,

```
git rebase -i
<show example of commits with possible flags>
```
I would draw attention to the `-f` (fixup) flag, which I almost always use rather than `-s` (squash).

File: source/dev_guide/project-standards.rst
Comment (forsyth2) [inline source/dev_guide/project-standards.rst:117]: It runs on pushing directly to `master`? Do we want to be encouraging that?

File: source/dev_guide/project-standards.rst
Comment (forsyth2) [inline source/dev_guide/project-standards.rst:123]: `Only if #2 succeeds AND tagging a release <link to how to tag a release>`

File: source/dev_guide/project-standards.rst
Comment (forsyth2) [inline source/dev_guide/project-standards.rst:154]: I think this might be better with "Writing..." and "Commenting..."

It's relatively easy to miss that this is a "AVOID" list, and the "Write/Comment" form makes it sound like a to-do list. (At least, that's how it reads to me)

File: source/install.rst
Comment (forsyth2) [inline source/install.rst:8]: I would replace "Cori" with "NERSC" too.

File: source/install.rst
Comment (forsyth2) [inline source/install.rst:171]: Do you have to clone your particular fork? I usually clone the main repo and then set my fork as a remote in addition to `origin` (i.e., the main repo, which it looks like you're calling `upstream`).

File: source/install.rst
Comment (forsyth2) [inline source/install.rst:190]: I would point out that this is where you'll see rebasing/merge conflicts if they exist.

**Review by forsyth2 — COMMENTED** [review]
Overall looks good, just a few comments.

File: source/dev_guide/project-standards.rst
Comment (tomvothecoder) [reply source/dev_guide/project-standards.rst:87]: Sounds good, as long as the isolated commits pass the build.

I'll also add `git bisect` to the list.

File: source/dev_guide/project-standards.rst
Comment (tomvothecoder) [reply source/dev_guide/project-standards.rst:117]: CI services have conditionals that will run the build on a `push` event to `master`, which is triggered from merging a PR. So in this context, we aren't pushing directly `master`. Also only the repo owner and added collaborators with write access can push to `master` (although we should probably add a branch protection rule to restrict it for everyone).

To make that description clearer, I will add "event" after ``push`` and ``pull_request``.

File: source/dev_guide/project-standards.rst
Comment (tomvothecoder) [reply source/dev_guide/project-standards.rst:154]: I can see how that might be confusing. I had **Don't** to prefix the sentence but removed it. I'll add it back in.

**Review by chengzhuzhang — APPROVED** [review]
Looking good. Thank you for taking time documenting! Just one comment/question: it might be handy to include `sphinx` in development env? if it is a small package.

**Comment by forsyth2** [comment]
> Looking good. Thank you for taking time documenting! Just one comment/question: it might be handy to include `sphinx` in development env? if it is a small package.

I agree, especially once #352 happens. It would be useful to not have to switch conda environments between testing code and checking the docs.

File: source/install.rst
Comment (tomvothecoder) [reply source/install.rst:171]: The common convention is to clone your fork and add the upstream repo to keep your fork `master` in sync with the upstream `master`. There are alternatives like the one you mentioned, but this method is most popular.

- https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#keep-your-fork-synced
- https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow

**Comment by tomvothecoder** [comment]
@chengzhuzhang @forsyth2 Thanks for reviewing. And yup, `sphinx` will be included in the dev env for #352.

File: source/install.rst
Comment (forsyth2) [reply source/install.rst:171]: Ah ok, good to know. Thanks

**Pull request merged**
